### ISSUE TITLE
Lack of general record member referencing

### ISSUE THREAD

**Issue by henrikt-ma** [modelica/ModelicaSpecification#3510]
In https://github.com/modelica/ModelicaSpecification/pull/3395 we added array subscripting of general expressions, but was there a reason for not also doing the analog for records?  In System Modeler we have both variants in our internal representation, and in the perspective of test implementing the Base Modelica proposal it seems we would need general record member referencing just as much as general array subscripting.

**Comment by HansOlsson** [comment]
> I suppose one would define general record member referencing primarily as an operation on a non-array record expression, and then specify that vectorization applies. I don't think this would lead to the same risk of ambiguity as for array subscripting, but mandatory parentheses might still be a good idea for the sake of syntax consistency. That is, instead of simply writing `f().a` (relying on a higher precedence of function application than record subscripting), one would write `(f()).a`.

Agreed, and that also keeps/opens up the possibility to use `f().a` to refer to the a-output of the (multi-returning) function `f()`.
